clients/go/pkg/zbc/client_test.go

```diff
@@ -544,6 +545,60 @@ func (s *clientTestSuite) TestOverrideHostAndPortEnvVar() {
 	s.EqualValues(fmt.Sprintf("%s:%s", address, port), config.GatewayAddress)
 }
 
+func (s *clientTestSuite) TestRetryPredicatewithStopOnPermanentError() {
+	var alwaysRetryfunc = func(_ context.Context, _ error) bool {
+		return true
+	}
+
+	for _, tc := range []struct {
+		err   error
+		retry bool
+	}{
+		{
+			err:   nil,
+			retry: true,
+		},
+		{
+			err:   Error("random error"),
+			retry: true,
+		},
+		{
+			err:   context.Canceled,
+			retry: false,
+		},
+		{
+			err:   context.DeadlineExceeded,
+			retry: false,
+		},
+		{
+			err:   status.Error(codes.InvalidArgument, "invalid arguments"),
+			retry: false,
+		},
+		{
+			err:   status.Error(codes.Unimplemented, "uninplemented"),
+			retry: false,
+		},
+		{
+			err:   status.Error(codes.Canceled, "canceled"),
+			retry: false,
+		},
+		{
+			err:   status.Error(codes.NotFound, "not found"),
+			retry: true,
```

Comment:
💭 Wondering whether it makes sense to retry this or shouldn't we also stop here.

Reply:
I feel like not found is generally considered transient, something which is not found now could be found later.

Reply:
We do have some cases where it's not transient. For example, if you activate a job and want to complete it. Job activation works with at-least-once semantics, so it's possible for two workers to see the same job. If both try to complete the job, one will get an `OK`, and the a `NOT_FOUND`. There's no point retrying the `NOT_FOUND` here.

Conversely, there's the case of process deployments. If you deploy a process, you get an answer as soon as the first partition processed it, but before it's been fully distributed to all partitions. This means if you try to create an instance for it immediately on a different partition, you would get `NOT_FOUND`, and in this case it _is_ transient.

So I guess it's a case-by-case based on the command here =/